engine-nextgen/src/main/kotlin/graphql/nadel/NextgenEngine.kt

```diff
@@ -32,7 +33,9 @@ import kotlinx.coroutines.future.asCompletableFuture
 import kotlinx.coroutines.future.asDeferred
 import java.util.concurrent.CompletableFuture
 
-class NextgenEngine(nadel: Nadel) : NadelExecutionEngine {
+class NextgenEngine(nadel: Nadel, transforms: List<NadelTransform<Any>> = emptyList()) : NadelExecutionEngine {
+    constructor(nadel: Nadel) : this(nadel, emptyList())
```

Comment:
additional constructor for Java compatibility. If I understand it correctly, Java does not understand Kotlin's default constructor arguments, so without this constructor java will need to do `new NextgenEngine(nadel, List.of())`

Reply:
Ah you have to add `@JvmOverloads` and the Kotlin compiler will add the extra constructors with the default arguments.

You'll have to make the constructor keyword explicit

```
class Nadel @JvmOverloads constructor(nadel: Nadel ...
```